charts/agent-control/templates/install.yaml

```diff
@@ -0,0 +1,40 @@
+{{- if (not (index .Values "agent-control-deployment").disableRemoteUpdate) -}}
+{{- $installJobName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "install-job" ) -}}
+{{- $releaseName := printf "%s-deployment" .Release.Name -}}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  annotations:
+    helm.sh/hook: post-install,post-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+  name: {{ $installJobName }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  template:
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ include "newrelic.common.naming.truncateToDNSWithSuffix" (dict "name" .Release.Name "suffix" "install-job") }}
+      containers:
+        - name: install-agent-control-deployment
+          # TODO: Pin a version to avoid problems with breaking changes.
+          image: newrelic/newrelic-agent-control-cli
```

Comment:
I think we should pin the cli version and update it manually. If not, an update to the cli could break the code of existing clients. However, we don't have any published version yet.

Reply:
Should a line in https://newrelic.atlassian.net/wiki/spaces/INST/pages/4056809473/version+Release be enough? Under Helm Charts.

```
- [ ] Update installation job image version
```

Reply:
You mean for integration tests and all that? Okay.

Reply:
Not only for integration tests but also for user needed a private registry/repository.

Reply:
I was thinking about prefixing with "dev", but if some users might need to use their private repositories, I'll leave it. Why would clients need the private repository?

I don't like that now the user can shoot his foot by putting whatever tag they want in there. When it was hardcoded, we could enforce a correct relation between the cli version and the chart version.

Reply:
I will still need to modify https://newrelic.atlassian.net/wiki/spaces/INST/pages/4056809473/version+Release to update the "tag" in the chart values, I assume.

Reply:
It actually depends on their requirements. I can imagine a security requirement where all images are mirrored into a private repository after some internal and independent security checks. I believe we have already received a work request to support such as use case (in regards of Agent Types where the repository is not configurable for now)